kotlinx-coroutines-core/common/src/flow/AbstractSharedFlow.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2016-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.flow
+
+import kotlinx.coroutines.assert
+import kotlinx.coroutines.channels.*
+import kotlinx.coroutines.flow.internal.*
+import kotlinx.coroutines.internal.*
+import kotlin.coroutines.*
+
+internal abstract class AbstractSharedFlowSlot<F> {
```

Comment:
Moved `AbstractSharedFlow`. I propose to leave `xxxImpl` classes in the `flow` package, though. Two reasons:
* We usually keep interface & impl in the same package (that's true for all other entities)
* People will see `xxxImpl` classes in debugger & stack-traces a lot. It would be clearer with shorter package names.